ql/src/java/org/apache/hadoop/hive/ql/optimizer/GenMapRedUtils.java

```diff
@@ -1720,6 +1723,41 @@ protected static MoveWork mergeMovePaths(Path condInputPath, MoveWork linkedMove
     return newWork;
   }
 
+  private static void setCustomStorageHandlerPropertiesForMerge(ConditionalResolverMergeFilesCtx mrCtx, MoveWork work) {
```

Comment:
that method sets both storageHandler and properties. btw what does it mean custom StorageHandler?

Reply:
Renamed it to `setCustomStorageHandlerAndPropertiesForMerge`.